commit fea2908c1cef
Author: J. Ryan Stinnett <jryans@gmail.com>
Date:   Thu Sep 21 16:43:37 2017 -0500

    Bug 1401256 - servo: Update inherited prop state for animation (from jryans:stylo-anim-inherit-context). r=heycam, a=sledru
    
    Source-Repo: https://github.com/servo/servo
    Source-Revision: 83705a8fa8992a974b32acc6635c7dfeed1afa50
    
    --HG--
    extra : source : 443fe7c5dd42bb6207fb778f58f94354d19fac28
---
 .../components/style/properties/helpers/animated_properties.mako.rs | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git servo/components/style/properties/helpers/animated_properties.mako.rs servo/components/style/properties/helpers/animated_properties.mako.rs
index ee3f601cdb91..5f98b7e5c0aa 100644
--- servo/components/style/properties/helpers/animated_properties.mako.rs
+++ servo/components/style/properties/helpers/animated_properties.mako.rs
@@ -576,6 +576,12 @@ impl AnimationValue {
             % for prop in data.longhands:
             % if prop.animatable:
             PropertyDeclaration::${prop.camel_case}(ref val) => {
+                context.for_non_inherited_property =
+                    % if prop.style_struct.inherited:
+                        None;
+                    % else:
+                        Some(LonghandId::${prop.camel_case});
+                    % endif
             % if prop.ident in SYSTEM_FONT_LONGHANDS and product == "gecko":
                 if let Some(sf) = val.get_system() {
                     longhands::system_font::resolve_system_font(sf, context);

commit 8a2e8131e3a9
Author: J. Ryan Stinnett <jryans@gmail.com>
Date:   Wed Sep 20 13:11:40 2017 -0500

    Bug 1401256 - Crashtest for animating lengths. r=heycam, a=sledru
    
    MozReview-Commit-ID: HiVfx9cg4ZZ
    
    --HG--
    extra : source : 34d60bb9e16bfc33223e9616ad3b6d664cb6d93a
---
 layout/style/crashtests/1401256.html    | 5 +++++
 layout/style/crashtests/crashtests.list | 1 +
 2 files changed, 6 insertions(+)

diff --git layout/style/crashtests/1401256.html layout/style/crashtests/1401256.html
new file mode 100644
index 000000000000..2a25dedc7e6c
--- /dev/null
+++ layout/style/crashtests/1401256.html
@@ -0,0 +1,5 @@
+<script>
+  let o1 = document.createElement('p');
+  document.documentElement.appendChild(o1);
+  o1.animate({'minWidth':['-moz-max-content']});
+</script>
\ No newline at end of file
diff --git layout/style/crashtests/crashtests.list layout/style/crashtests/crashtests.list
index 9a1412863e6a..7466eb298a8f 100644
--- layout/style/crashtests/crashtests.list
+++ layout/style/crashtests/crashtests.list
@@ -202,3 +202,4 @@ load 1387481-1.html
 load 1387499.html
 load 1391577.html
 load 1398581.html
+load 1401256.html
